### PR TITLE
asNumber: return undefined when value is empty string

### PR DESCRIPTION
**Pull request by sjhewitt**
It is currently impossible to clear the value of a number field that is not required because `asNumber` converts the empty string to `0`. This change adds a special case that returns `undefined`, so that the input can be cleared

### Checklist

* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [x] I've updated docs if needed

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -223,6 +223,9 @@ export function mergeObjects(obj1, obj2, concatArrays = false) {
 }
 
 export function asNumber(value) {
+  if(value === "") {
```
Comment (n1k0) [inline src/utils.js]: Please add a space after `if`. Would be nice to add an eslint rule for this if you have the time btw :)

**Comment by sjhewitt** [comment]
@n1k0 - done, and I added the linter rule and fixed any errors

**Comment by n1k0** [comment]
@sjhewitt Fantastic, thank you so much :+1:

**Comment by n1k0** [comment]
Released in [v0.41.0](https://github.com/mozilla-services/react-jsonschema-form/releases/tag/v0.41.0).

**Pull request merged**